### DISCUSSION TITLE
Long delay with crazyflie 2.1 and  flow deck v2

### DISCUSSION THREAD

**Discussion by sanjuchopracool** [bitcraze/discussions#287] [General]
I just got a crazyflie 2.1 and flow deck v2. I updated firmware and started using command based flight control.
There is a long delay between take off button click and actual takeoff and same with land also.

In log settings->log configurations,  I have selected 100 ms (tried with 100ms also)
I have tried with VM machine (4 cpu 4GB RAM 128 mb Graphics), I also tried with native Arch Linux (connect/disconnect is much faster)

There is around 3-5 sec delay, sometime more in landing.

cfclient almost become unresponsive during flight.
Is this delay normal for cfclient or I am missing some setting ?

**Comment by krichardsson** [comment]
I think the implementation of the "Command based flight control" has some flaws and I'm not surprised that you see delays. I have seen it too and it should be fixed at some point. 

Do you see the same problems if you run an example, [the motion_commander_demo](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/autonomy/motion_commander_demo.py) for instance?

We have had problems with performance in the client from time to time on some platforms, maybe this is what you see?

**Reply by sanjuchopracool** [reply]
Motion commander seems to work fine, but cfclient become unresponsive after a while of connect. Another thing I noticed is, I changed logging time from 20 to 1000 ms. But after on and off, logging setting got reverted back to 20 ms, looks like it does not persist. Which parameter I can check for logging time ?

**Reply by krichardsson** [reply]
I'm not exactly sure what you mean with "logging time" in this case. Do you mean the "logging time" in the "Log configuration" dialog window?

This time is stored in each log configuration (they can all have different times) and are sent to the Crazyflie when you select the log configuration in the log tab.

**Comment by krichardsson** [comment]
We have also sometimes had performance issues in the client through the years depending on OS and sometimes for various QT versions.

**Comment by krichardsson** [comment]
Have you tried to use the example scripts in [crazyflie-lib-python](https://github.com/bitcraze/crazyflie-lib-python)? Do you see the same delay here or is it only in the client?
